app/src/main/java/com/amaze/filemanager/filesystem/EditableFileAbstraction.java

```diff
@@ -57,23 +57,27 @@ public EditableFileAbstraction(@NonNull Context context, @NonNull Uri uri) {
         this.scheme = CONTENT;
 
         String tempName = null;
-        Cursor c =
-            context
-                .getContentResolver()
-                .query(uri, new String[] {OpenableColumns.DISPLAY_NAME}, null, null, null);
-
-        if (c != null) {
-          c.moveToFirst();
-          try {
-            /*
-            The result and whether [Cursor.getString()] throws an exception when the column
-            value is null or the column type is not a string type is implementation-defined.
-            */
-            tempName = c.getString(c.getColumnIndex(OpenableColumns.DISPLAY_NAME));
-          } catch (Exception e) {
-            tempName = null;
+        try {
+          Cursor c =
+              context
+                  .getContentResolver()
+                  .query(uri, new String[] {OpenableColumns.DISPLAY_NAME}, null, null, null);
+
+          if (c != null) {
+            c.moveToFirst();
+            try {
+              /*
+              The result and whether [Cursor.getString()] throws an exception when the column
+              value is null or the column type is not a string type is implementation-defined.
+              */
+              tempName = c.getString(c.getColumnIndex(OpenableColumns.DISPLAY_NAME));
+            } catch (Exception e) {
+              tempName = null;
+            }
+            c.close();
           }
-          c.close();
+        } catch (UnsupportedOperationException e) {
+          tempName = uri.getLastPathSegment();
```

Comment:
What's this for?

Reply:
The catch I mean, why is it there.

Reply:
Yes of course. The problem is, I even forgotten why I inserted this catch at the beginning 😵️ must be to prevent some empty cursor I think.

This is a good example of what a good programmer should do - document as you code.